### ISSUE TITLE
Não imprime qr code Epson TM-T20?

### ISSUE THREAD

**Issue by dasiolacerda** [nfephp-org/posprint#85]
Ate onde vi estou fazendo tudo certo, mas não estou conseguindo imprimir o qrcode a impressora Epson, eis o que esta gerando na hora que executa o código.
[ESC] ! [NUL] [ESC] M (01h) [ESC] a (01h) [ESC] d (01h) Consulte via Leitor de QRCode [ESC] d (01h) ---------------------------------------------------------------- [GS] (k [EOT] [NUL] 1A1 [NUL] [GS] (k (03h) [NUL] 1C [ENQ] [GS] (k (03h) [NUL] 1E0 [GS] (k [EM] [NUL] 1P0TESTE DE GERAR QR-CODE [GS] (k (03h) [NUL] 1Q0 [ESC] E (01h) [ESC] d (01h) ---------------------------------------------------------------- NOTA FISCAL INVALIDA - SEM PROTOCOLO DE AUTORIZACAO [ESC] d (01h) ---------------------------------------------------------------- [ESC] ! [NUL] [ESC] M (01h) [ESC] d (03h) [GS] VB (03h) 

se alguém passou pelo mesmo problema puder ajudar,

**Comment by robmachado** [comment]
Esse meu amigo é o código renderizado para VISUALIZAÇÃO onde os comandos binários foram convertidos em TEXTOS para serem visualizados e não é o código que deve ir para impressora

**Comment by dasiolacerda** [comment]
Sim, estou ciente disso, esta vindo assim, e sei que não é isso que é enviado mesmo, a questão é que esta um pouco diferente essa visualização de outro sistema que ja tinha aqui na empresa, e do sistema funciona perfeitamente, e esse aqui não funciona de jeito nenhum, e preciso fazer a migração do sistema desktop pra esse online.
Ultima coisa que falta é o QRcode, e não acho solução de jeito nenhum.

**Comment by robmachado** [comment]
O problema que eu enfrento é que não tenho acesso as impressoras para fazer os testes, então todo o trabalho foi feito com base exclusivamente nos manuais das impressoras. O que realmente não é suficiente para garantir o funcionamento. Não recebo nenhum tipo de ajuda de quem tem as impressoras para prover os testes e correções necessárias.

**Comment by 007amauri** [comment]
Seria alguma coisa assim em javascript

function Bar_Code39(){ // Bar Cod Commands
this.Texto = this.Texto+String.fromCharCode(29)+"kE";// Modelo Bar Code39
this.Texto = this.Texto+String.fromCharCode(29)+"H2";//Seleciona a posição de impressão de caracteres HRI 
this.Texto = this.Texto+String.fromCharCode(29)+"f0";//Seleciona a fonte para os caracteres HRI
this.Texto = this.Texto+String.fromCharCode(29)+"w2"; //Definir a largura da barra do código de barras.
this.Texto = this.Texto+String.fromCharCode(29)+"h32";//Ajuste a altura do código de barra a n pontos.
}
function CORTAR(){ 
this.Texto = this.Texto+String.fromCharCode(27);
this.Texto = this.Texto+String.fromCharCode(109); 
} 

function Impr(){ 
this.Linha = 0 ; 
this.Coluna = 0 ; 
this.Texto = "" ; 
this.Porta = "LPT1" ; 
this.Arquivo = "c:\\"+this.Porta ; 
this.Code39 = Bar_Code39;

this.cortapapel = CORTAR;
}
